realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1092,6 +1093,29 @@ public boolean isValid() {
         return this;
     }
 
+    /**
+     * Returns a distinct set of objects of a specific class. As a Realm is unordered, it is
+     * undefined which objects are returned in case of multiple occurrences.
```

Comment:
From the core doc, this behaviour is defined i think? From doc of `TableView.distinct`: 
> If two rows are indentical (for the given set of distinct-columns), then the last row is removed.